### PR TITLE
Merge v0.23.9

### PR DESCRIPTION
**Pull request by schancel**

### CODE REVIEW

File: src/qt/overviewpage.cpp [hunk 1]
```diff
@@ -308,7 +308,7 @@ void OverviewPage::setWalletModel(WalletModel *model) {
                 });
     }
 
-    // update the display unit, to not use the default ("BCH")
+    // update the display unit, to not use the default ("XEC")
```
Comment (EyeOfPython) [inline src/qt/overviewpage.cpp:311]: XPI

File: src/qt/sendcoinsentry.cpp [hunk 2]
```diff
@@ -129,7 +129,7 @@ void SendCoinsEntry::clear() {
     ui->memoTextLabel_s->clear();
     ui->payAmount_s->clear();
 
-    // update the display unit, to not use the default ("BCH")
+    // update the display unit, to not use the default ("XEC")
```
Comment (EyeOfPython) [inline src/qt/sendcoinsentry.cpp:132]: XPI

File: src/qt/test/paymentrequestdata.h [hunk 3]
```diff
@@ -430,7 +430,8 @@ d2hj739GDLz0b5KuJ2SG6VknMRQM976w/m2qlq0ccVGaaZ2zMIGfpzL3p6adwx/5\
 ";
 
 //
-// Payment request with amount overflow (amount is set to 21000001 BCH)
+// Payment request with amount overflow (amount is set to
+// 21,000,001,000,000 XEC)
```
Comment (EyeOfPython) [inline src/qt/test/paymentrequestdata.h:434]: XPI (adjust amount)

File: src/rpc/mining.cpp [hunk 4]
```diff
@@ -527,7 +527,7 @@ static UniValue getmininginfo(const Config &config,
     return obj;
 }
 
-// NOTE: Unlike wallet RPC (which use BCH values), mining RPCs follow GBT (BIP
+// NOTE: Unlike wallet RPC (which use XEC values), mining RPCs follow GBT (BIP
```
Comment (EyeOfPython) [inline src/rpc/mining.cpp:530]: XPI

File: src/wallet/test/coinselector_tests.cpp [hunk 5]
```diff
@@ -528,14 +528,14 @@ BOOST_AUTO_TEST_CASE(knapsack_solver_test) {
         BOOST_CHECK(testWallet.SelectCoinsMinConf(
             95 * CENT, filter_confirmed, GroupCoins(vCoins), setCoinsRet,
             nValueRet, coin_selection_params, bnb_used));
-        // we should get 1 BCH in 1 coin
+        // we should get 1,000,000 XEC in 1 coin
```
Comment (EyeOfPython) [inline src/wallet/test/coinselector_tests.cpp:531]: XPI (adjust amount)

File: src/wallet/test/coinselector_tests.cpp [hunk 6]
```diff
@@ -528,14 +528,14 @@ BOOST_AUTO_TEST_CASE(knapsack_solver_test) {
...
         BOOST_CHECK_EQUAL(nValueRet, 1 * COIN);
         BOOST_CHECK_EQUAL(setCoinsRet.size(), 1U);
 
         BOOST_CHECK(testWallet.SelectCoinsMinConf(
             195 * CENT, filter_confirmed, GroupCoins(vCoins), setCoinsRet,
             nValueRet, coin_selection_params, bnb_used));
-        // we should get 2 BCH in 1 coin
+        // we should get 2,000,000 XEC in 1 coin
```
Comment (EyeOfPython) [inline src/wallet/test/coinselector_tests.cpp:538]: XPI (adjust amount)

File: test/functional/rpc_psbt.py [hunk 7]
```diff
@@ -103,7 +103,8 @@ def run_test(self):
         self.nodes[1].sendrawtransaction(
             self.nodes[1].finalizepsbt(walletprocesspsbt_out['psbt'])['hex'])
 
-        # feeRate of 0.1 BCH / KB produces a total fee slightly below -maxtxfee
+        # feeRate of 100,000 XEC / KB produces a total fee slightly below
```
Comment (EyeOfPython) [inline test/functional/rpc_psbt.py:106]: XPI (adjust amount)

File: test/functional/rpc_rawtransaction.py [hunk 8]
```diff
@@ -608,7 +612,7 @@ def run_test(self):
         rawTx = self.nodes[2].createrawtransaction(inputs, outputs)
         rawTxSigned = self.nodes[2].signrawtransactionwithwallet(rawTx)
         assert_equal(rawTxSigned['complete'], True)
-        # Fee 10,000 satoshis, ~200 b transaction, fee rate should land around 50 sat/byte = 0.00050000 BCH/kB
+        # Fee 10,000 satoshis, ~200 b transaction, fee rate should land around 50 sat/byte = 500 XEC/kB
```
Comment (EyeOfPython) [inline test/functional/rpc_rawtransaction.py:615]: XPI (adjust amount)

File: test/functional/rpc_rawtransaction.py [hunk 9]
```diff
@@ -640,7 +644,7 @@ def run_test(self):
         rawTx = self.nodes[2].createrawtransaction(inputs, outputs)
         rawTxSigned = self.nodes[2].signrawtransactionwithwallet(rawTx)
         assert_equal(rawTxSigned['complete'], True)
-        # Fee 2,000,000 satoshis, ~100 b transaction, fee rate should land around 20,000 sat/byte = 0.20000000 BCH/kB
+        # Fee 2,000,000 satoshis, ~100 b transaction, fee rate should land around 20,000 sat/byte = 200,000 XEC/kB
```
Comment (EyeOfPython) [inline test/functional/rpc_rawtransaction.py:647]: XPI (adjust amount)

File: test/functional/test_framework/util.py [hunk 10]
```diff
@@ -48,10 +48,10 @@ def assert_fee_amount(fee, tx_size, fee_per_kB, wiggleroom=2):
     target_fee = round(tx_size * fee_per_kB / 1000, 8)
     if fee < (tx_size - wiggleroom) * fee_per_kB / 1000:
         raise AssertionError(
-            "Fee of {} BCH too low! (Should be {} BCH)".format(str(fee), str(target_fee)))
+            "Fee of {} XEC too low! (Should be {} XEC)".format(str(fee), str(target_fee)))
```
Comment (EyeOfPython) [inline test/functional/test_framework/util.py:51]: XPI

File: test/functional/test_framework/util.py [hunk 11]
```diff
@@ -48,10 +48,10 @@ def assert_fee_amount(fee, tx_size, fee_per_kB, wiggleroom=2):
...
     if fee > (tx_size + wiggleroom) * fee_per_kB / 1000:
         raise AssertionError(
-            "Fee of {} BCH too high! (Should be {} BCH)".format(str(fee), str(target_fee)))
+            "Fee of {} XEC too high! (Should be {} XEC)".format(str(fee), str(target_fee)))
```
Comment (EyeOfPython) [inline test/functional/test_framework/util.py:54]: XPI

File: test/functional/wallet_abandonconflict.py [hunk 12]
```diff
@@ -89,15 +89,15 @@ def total_fees(*txids):
             self.nodes[0].createrawtransaction(inputs, outputs))
         txAB1 = self.nodes[0].sendrawtransaction(signed["hex"])
 
-        # Identify the 14.99998btc output
+        # Identify the 14,999,980 XEC output
```
Comment (EyeOfPython) [inline test/functional/wallet_abandonconflict.py:92]: XPI (adjust amount)

File: test/functional/wallet_abandonconflict.py [hunk 13]
```diff
@@ -89,15 +89,15 @@ def total_fees(*txids):
...
         nAB = next(tx_out["vout"] for tx_out in self.nodes[0].gettransaction(
             txAB1)["details"] if tx_out["amount"] == Decimal('1499.998'))
 
         # Create a child tx spending AB1 and C
         inputs = []
-        # Amount 14.99998 BCH
+        # Amount 14,999,980 XEC
```
Comment (EyeOfPython) [inline test/functional/wallet_abandonconflict.py:98]: XPI (adjust amount)

File: test/functional/wallet_abandonconflict.py [hunk 14]
```diff
@@ -89,15 +89,15 @@ def total_fees(*txids):
...
         inputs.append({"txid": txAB1, "vout": nAB})
-        # Amount 10 BCH
+        # Amount 10,000,000 XEC
```
Comment (EyeOfPython) [inline test/functional/wallet_abandonconflict.py:100]: XPI (adjust amount)

File: test/functional/wallet_abandonconflict.py [hunk 15]
```diff
@@ -203,15 +203,16 @@ def total_fees(*txids):
         connect_nodes(self.nodes[0], self.nodes[1])
         self.sync_blocks()
 
-        # Verify that B and C's 10 BCH outputs are available for spending again
-        # because AB1 is now conflicted
+        # Verify that B and C's 10,000,000 XEC outputs are available for
```
Comment (EyeOfPython) [inline test/functional/wallet_abandonconflict.py:206]: XPI (adjust amount)

File: test/functional/wallet_abandonconflict.py [hunk 16]
```diff
@@ -203,15 +203,16 @@ def total_fees(*txids):
...
         newbalance = self.nodes[0].getbalance()
         assert_equal(newbalance, balance + Decimal('2000'))
         balance = newbalance
 
-        # There is currently a minor bug around this and so this test doesn't work.  See Issue #7315
-        # Invalidate the block with the double spend and B's 10 BCH output should no longer be available
-        # Don't think C's should either
+        # There is currently a minor bug around this and so this test doesn't
+        # work.  See Issue #7315
+        # Invalidate the block with the double spend and B's 10,000,000 XEC
```
Comment (EyeOfPython) [inline test/functional/wallet_abandonconflict.py:214]: XPI (adjust amount)

File: test/functional/wallet_avoidreuse.py [hunk 17]
```diff
@@ -324,7 +324,7 @@ def test_sending_from_reused_address_fails(self):
         # getbalances should show 1000 used, 500 Lotus trusted
         assert_balances(self.nodes[1], mine={"used": 1000, "trusted": 500})
 
-        # node 1 should now have a balance of 5 (no dirty) or 15 (including
+        # node 1 should now have a balance of 5MM (no dirty) or 15MM (including
```
Comment (EyeOfPython) [inline test/functional/wallet_avoidreuse.py:327]: adjust amount to XPI

File: test/functional/wallet_import_rescan.py [hunk 18]
```diff
@@ -8,12 +8,12 @@
 importmulti RPCs with different types of keys and rescan options.
 
 In the first part of the test, node 0 creates an address for each type of
-import RPC call and node 0 sends BCH to it. Then other nodes import the
+import RPC call and node 0 sends XEC to it. Then other nodes import the
```
Comment (EyeOfPython) [inline test/functional/wallet_import_rescan.py:11]: XPI

File: test/functional/wallet_listsinceblock.py [hunk 19]
```diff
@@ -150,8 +150,8 @@ def test_double_spend(self):
 
         Problematic case:
 
-        1. User 1 receives BCH in tx1 from utxo1 in block aa1.
-        2. User 2 receives BCH in tx2 from utxo1 (same) in block bb1
+        1. User 1 receives XEC in tx1 from utxo1 in block aa1.
```
Comment (EyeOfPython) [inline test/functional/wallet_listsinceblock.py:153]: XPI

**Review by EyeOfPython — CHANGES_REQUESTED** [review]

File: src/node/context.cpp [hunk 20]
```diff
@@ -9,6 +9,7 @@
 #include <net.h>
 #include <net_processing.h>
 #include <scheduler.h>
+#include <txmempool.h>
```
Comment (schancel) [inline src/node/context.cpp:12]: @EyeOfPython This seems unnecessary?

File: contrib/aur/bitcoin-abc-qt/PKGBUILD [hunk 21]
```diff
@@ -1,7 +1,7 @@
 # Maintainer: Josh Ellithorpe <quest@mac.com>
 
 pkgname=bitcoin-abc-qt
-pkgver=0.23.8
+pkgver=0.23.9
```
Comment (schancel) [inline contrib/aur/bitcoin-abc-qt/PKGBUILD:4]: Guess this file goes back further. Needs to be deleted.

Comment (schancel) [reply contrib/aur/bitcoin-abc-qt/PKGBUILD:4]: Will delete before release 1.0.3. Merges are annoying to rebase.

File: src/qt/test/paymentservertests.cpp [hunk 22]
```diff
@@ -205,7 +205,7 @@ void PaymentServerTests::paymentServerTests() {
     // compares 50001 <= BIP70_MAX_PAYMENTREQUEST_SIZE == false
     QCOMPARE(PaymentServer::verifySize(tempFile.size()), false);
 
-    // Payment request with amount overflow (amount is set to 21000001 BCH):
+    // Payment request with amount overflow (amount is set to 21000001 XPI):
```
Comment (EyeOfPython) [inline src/qt/test/paymentservertests.cpp:208]: Shouldn't it be 2100000100 XPI?

**Review by EyeOfPython — APPROVED** [review]

**Pull request merged**